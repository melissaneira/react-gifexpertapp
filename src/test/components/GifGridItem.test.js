import { shallow } from 'enzyme';
import GifGridItem from '../../Components/GifGridItem';
describe('GifGridItem Component', () => {

    const title = "Nueva Imagen";
    const url = 'https://localhost:8080/mi_imagen';
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);

    test('Probar que componente se renderize correctamente ', () => {       
        expect(wrapper).toMatchSnapshot();    
    })    

    test('debe de tener un parrafo con el title ', () => {
        const p = wrapper.find('p');
        expect(p.text()).toBe(title);
        console.log(p.text());
    })

    test('debe de tener la imagen igual al url y alt', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);        
    })

    test('debe tener animate_fadeIn', () => { 
        const div = wrapper.find('div');    
        console.log(div.hasClass('animate__fadeIn'));
        expect(div.hasClass('animate__fadeIn')).toBe(true);       
    })
})