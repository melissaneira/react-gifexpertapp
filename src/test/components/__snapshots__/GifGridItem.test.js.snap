// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem Component Probar que componente se renderize correctamente  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Nueva Imagen"
    src="https://localhost:8080/mi_imagen"
  />
  <p>
    Nueva Imagen
  </p>
</div>
`;
